### PR TITLE
Fixes #756 Embedded Entity Migration

### LINKED ISSUE #756: Embedded entity migration

### PR DESCRIPTION
**Pull request by tadean**
This provides a process plugin for embedded entities that parses HTML and translates embedded entity IDs according to migration lookups.

## Description
Fixes #756 

## Related Issue
#756 

## How Has This Been Tested?
- Create text with embedded entities in a source d7 system
- Verify if embedded entities are migrated into d9 when migrations are run

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I've added this Pull Request to the AZ Quickstart Github project and set it to "Review in progress".
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by tadean** [comment]
Needs more testing, and discussion of view mode mapping.

**Comment by joeparsons** [comment]
This is looking great.  Nice work @tadean!

File: modules/custom/az_migration/src/Plugin/migrate/process/EntityEmbedProcess.php
Comment (joeparsons) [inline modules/custom/az_migration/src/Plugin/migrate/process/EntityEmbedProcess.php:14]: I think we should probably add some inline documentation before merging. (At least a list of config options, if any, and a code example)

**Comment by camikazegreen** [comment]
I could not figure out how to get this to work.

File: modules/custom/az_migration/src/Plugin/migrate/process/EntityEmbedProcess.php
Comment (tadean) [reply modules/custom/az_migration/src/Plugin/migrate/process/EntityEmbedProcess.php:14]: Added @joeparsons suggestion of documentation of usage as well as addition of configuration options that allow for specifying and overriding information for custom content migrations.

**Comment by tadean** [comment]
Probo migration build page with embedded content: https://a37475c0-fc7d-4543-b0ed-814994fb56bf--site-migration.probo.build/news/2022/02/black-holes-come-big-screen

**Review by trackleft — APPROVED** [review]

File: modules/custom/az_migration/src/Plugin/migrate/process/EntityEmbedProcess.php
Comment (joeparsons) [inline modules/custom/az_migration/src/Plugin/migrate/process/EntityEmbedProcess.php:211]: Looks like calling `DOMDocument::loadHTML()` statically is deprecated and not supported in PHP 8.0.0+:
https://www.php.net/manual/en/domdocument.loadhtml.php#refsect1-domdocument.loadhtml-errors

File: modules/custom/az_migration/src/Plugin/migrate/process/EntityEmbedProcess.php
Comment (joeparsons) [inline modules/custom/az_migration/src/Plugin/migrate/process/EntityEmbedProcess.php:266]: Nitpick: could we avoid the calls to `Database::setActiveConnection()` by passing an argument to [`Database::getConnection()`](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Database%21Database.php/function/Database%3A%3AgetConnection/9.3.x)?

```suggestion
          // Lookup of content type.
          $node_type = Database::getConnection('migrate')
            ->query('SELECT type FROM {node} WHERE nid = :nid', [':nid' => $id])
            ->fetchField();
```

Comment (tadean) [reply modules/custom/az_migration/src/Plugin/migrate/process/EntityEmbedProcess.php:266]: That seems reasonable. I was a bit leery of altering the active connection. A one-time connection that doesn't alter anything seems better. Thanks @joeparsons !

**Review by joeparsons — DISMISSED** [review]

**Review by joeparsons — APPROVED** [review]

**Review by trackleft — APPROVED** [review]
I approve this, but wonder if we should follow the new return type declaration standard for drupal 9? https://www.drupal.org/docs/develop/standards/coding-standards#s-parameter-and-return-typehinting

**Comment by joeparsons** [comment]
> I approve this, but wonder if we should follow the new return type declaration standard for drupal 9? [drupal.org/docs/develop/standards/coding-standards#s-parameter-and-return-typehinting](https://www.drupal.org/docs/develop/standards/coding-standards#s-parameter-and-return-typehinting)

Wouldn't be opposed that change happening now but I think we could also address that as a followup (or potentially as part of a larger issue to apply that new coding standard everywhere...).

**Comment by trackleft** [comment]
Ok, let's make that a policy for 2.2.0

**Review by camikazegreen — APPROVED** [review]
Verified in Probo that it's still working on the only example that I know about for embedded entities in our migration example: https://f922cf2e-e088-4969-83a2-7b22e752d739--site-migration.probo.build/news/2022/02/black-holes-come-big-screen 

I'm not seeing any regressions.

**Pull request merged**
